base/docker-integration.go

```diff
@@ -0,0 +1,320 @@
+package base
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/couchbase/gocb"
+	"github.com/ory/dockertest"
+	"github.com/ory/dockertest/docker"
+)
+
+//Structs used when checking health of nodes and buckets
+type nodes struct {
+	Status string `json:"status"`
+	Uptime string `json:"uptime"`
+}
+
+type responseBody struct {
+	Nodes []nodes `json:"nodes"`
+}
+
+//This will start up a Docker Instance of Couchbase Server and will be used to run tests against
+func NewDockerTest(t *testing.M) {
+
+	//Only spin up a Docker Instance if environment variable is set
+	if os.Getenv("SG_TEST_USE_DOCKER") == "true" {
```

Comment:
Reversing this logic will alow you to remove a level of indentation for mostly the whole file. e.g:

```go
if os.Getenv("SG_TEST_USE_DOCKER") != "true" {
	os.Exit(t.Run())
}

// all the other docker code
```